From f6d7a149b40a7b131852f4daf8b428bbc993b12e Mon Sep 17 00:00:00 2001
From: Steve Langasek <steve.langasek@ubuntu.com>
Date: Sun, 8 Jul 2018 12:51:22 +0100
Subject: [PATCH 1/1] Compatibility with libgit2 0.27

git_config_add_file_ondisk() now takes an additional (optional) 'repo'
argument.  Fix the build failure.

[smcv: Make the API change conditional on LIBGIT2_SOVERSION so we can
build against either version]

Fixes: #1
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=903146
Signed-off-by: Simon McVittie <smcv@debian.org>
---
 libgit2-glib/ggit-config.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libgit2-glib/ggit-config.c b/libgit2-glib/ggit-config.c
index b6536ce..acdad95 100644
--- a/libgit2-glib/ggit-config.c
+++ b/libgit2-glib/ggit-config.c
@@ -295,6 +295,9 @@ ggit_config_add_file (GgitConfig      *config,
 	ret = git_config_add_file_ondisk (_ggit_native_get (config),
 	                                  path,
 	                                  (git_config_level_t)level,
+#if LIBGIT2_SOVERSION >= 27
+	                                  NULL,
+#endif
 	                                  force);
 	g_free (path);
 
-- 
2.14.2.windows.3

